platform_versioned_docs/version-24.1/data_studios/custom-envs.mdx

```diff
@@ -0,0 +1,143 @@
+name: example-env
+channels:
+  - defaults
+  - conda-forge
+dependencies:
+  - python=3.7
+  - numpy
+  - pandas
+  - scikit-learn
+  - pip:
+    - matplotlib
+    - seaborn
+    - jupyter
+```
+
+To create a data studio with custom Conda packages, see [Add a data studio][add-ds].
+
+## Custom container template image {#custom-containers}
+
+For advanced use cases, you can build your own container template image.
+
+:::note
+Amazon Elastic Container Registry (ECR) is the only supported private container registry.
+:::
+
+### Prerequisites
+
+- Access to a container image repository, either a public container registry or in a private Amazon ECR repository
+- A container template image
+
+### Dockerfile configuration {#dockerfile}
+
+For your custom template container image, you must use a Seqera provided base image and include several additional build steps for compatibility with Data Studios.
+
+#### Ports
+
+The container must use the value of the `CONNECT_TOOL_PORT` environment variable as the listening port for whatever interactive software that you include in your custom container.
+
+#### Signals
+
+Upon termination, the container's main process must handle the `SIGTERM` signal and perform any necessarily cleanup. After a 30 second grace period, the container receives the `SIGKILL` signal.
+
+#### Minimal Dockerfile
+
+The minimal Dockerfile includes directives to accomplish the following:
+
+- Pulls a Seqera-provided base image with prerequisite binaries
+- Installs prerequisite packages, such as wget and fuse
+- Installs and configures Seqera Fusion
+- Sets the container entry points
+
+Customize the following Dockerfile to include any additional software that you require:
+
+```docker title="Minimal Dockerfile"
+# Seqera base image
+FROM public.cr.seqera.io/platform/connect-client:0.7.1 as connect
+
+# 1. Add connect binary
+COPY --from=connect /connect-client /usr/bin/connect-client
+
+# 2. Install required system packages
+RUN apt-get update --yes && apt-get install --yes --no-install-recommends \
+    ca-certificates \
+    wget \
+    fuse \
+    btrfs-progs \
+    squashfs-tools \
+    && apt-get clean && rm -rf /var/lib/apt/lists/*
+
+# 3. Install Fusion
+RUN wget https://fusionfs.seqera.io/releases/pkg/2/4/2/fusion-amd64.tar.gz && \
+    tar xvzf fusion-amd64.tar.gz && \
+    rm fusion-amd64.tar.gz
+
+# 4. Configure Fusion env variables
+ENV FUSION_LOG_OUTPUT=/tmp/.fusion.log
+ENV FUSION_COMMAND_BUFFER_SIZE=0
+ENV FUSION_VALIDATE_WORK=false
+
+# 5. Configure connect and Fusion as the entrypoint
+ENTRYPOINT ["/usr/bin/fusion", "/usr/bin/connect-client"]
+```
+
+To run a basic Python-based HTTP server, build a container from the following Dockerfile. When a data studio runs the custom template environment, the value for the `CONNECT_TOOL_PORT` environment variable is provided dynamically.
+
+```docker title="Example Dockerfile with Python HTTP server"
+FROM public.cr.seqera.io/platform/connect-client:0.7.1 as connect
```

Comment:
The latest image should be: `cr.seqera.io/public/connect-client:0.7.4`, but we may want to allow some wiggle for security and minor updates, so unless there are objections @robnewman, we should use:

`cr.seqera.io/public/connect-client:0.7`

Reply:
Sounds good - agreed!